core/pbcc_subscribe_event_listener.c

```diff
@@ -248,6 +265,17 @@ static pbarray_t* pbcc_initialize_array_(
 //                 Functions
 // ----------------------------------------------
 
+pbcc_status_listener_t* pbcc_status_listener_alloc_(
+    pubnub_subscribe_status_callback_t callback,
+    void*                              user_data)
+{
+    PBCC_ALLOCATE_TYPE(listener, pbcc_status_listener_t, true, NULL);
+    listener->callback  = callback;
+    listener->user_data = user_data;
+
+    return listener;
+}
+
```

Comment:
Do we actually need this? 

Can we go in other way and do the following:
1. for `pubnub_subscribe_listener_type` add `PBSL_LISTENER_ON_STATUS`
2. Modify `pbcc_event_listener_add_status_listener` and `pbcc_event_listener_remove_status_listener`?

Also we will probably will need to make a change to: 
```c
pbcc_initialize_array_(&listener->global_status, NULL))
```
and add destructor call:
```c
pbcc_initialize_array_(&listener->global_status, pbcc_listener_free_))
```

Reply:
Hmm, but then we're mixing two different types of API and I'm not sure about it ;o